### DISCUSSION TITLE
Create orphan branch using octokit api #1984

### DISCUSSION THREAD

**Discussion by techierishi** [community/community#24699] [Apps, API and Webhooks]
<p>Is there an API to create an orphan branch using Octokit.<br>
Most probably and alternative of following git operation:</p>
<pre><code class="lang-bash">git checkout --orphan orp;
git commit -m 'orp branch initial commit';
git push -u origin orp;
</code></pre>
<aside class="onebox githubissue">
  <header class="source">
      <a href="https://github.com/octokit/rest.js/issues/1984" target="_blank" rel="noopener nofollow ugc">github.com/octokit/rest.js</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/octokit/rest.js/issues/1984" target="_blank" rel="noopener nofollow ugc">Create orphan branch using octokit api</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2021-01-20" data-time="12:23:35" data-timezone="UTC">12:23PM - 20 Jan 21 UTC</span>
      </div>

      <div class="user">
        <a href="https://github.com/techierishi" target="_blank" rel="noopener nofollow ugc">
          <img alt="techierishi" src="https://user-images.githubusercontent.com/7880021/181086312-3869c6c4-9807-44eb-84b5-a5eb8a15a960.jpeg" class="onebox-avatar-inline" width="20" height="20">
          techierishi
        </a>
      </div>
    </div>
  </div>
</div>

<div class="github-row">
  <p class="github-content">Is there an API to create an orphan branch using Octokit.
Most probably and alternative of following git operation:
git checkout --orphan orp;
git...</p>
</div>

<div class="labels">
    <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">support</span>
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by Leko** [comment, marked as answer]
<p>Hello, <a class="mention" href="/techierishi">@techierishi</a>. Is this a solution for you?<br>
Inspired by <a href="https://gist.github.com/jamezrin/fec71ea0e50969804e0ad46fe7737b88" class="inline-onebox" rel="noopener nofollow ugc">Python script to create an orphan branch · GitHub</a> and <a href="https://gist.github.com/jamezrin/fec71ea0e50969804e0ad46fe7737b88#gistcomment-3569582" rel="noopener nofollow ugc">the suggestion in the gist</a></p>
<pre><code class="lang-auto">// &gt; Git has a well-known, or at least sort-of-well-known, empty tree whose SHA1 ...
// https://stackoverflow.com/questions/9765453
const SHA1_EMPTY_TREE = "4b825dc642cb6eb9a060e54bf8d69288fbee4904";

const owner = 'xxx'
const repo = 'yyy'

const octokit = new Octokit({ auth: token });
const res = await octokit.request("POST /repos/{owner}/{repo}/git/commits", {
  owner,
  repo,
  message: "orp branch initial commit",
  tree: SHA1_EMPTY_TREE,
  parents: [],
});
await octokit.request("POST /repos/{owner}/{repo}/git/refs", {
  owner,
  repo,
  // If it doesn't start with 'refs' and have at least two slashes, it will be rejected.
  ref: 'refs/heads/orp',
  sha: res.data.sha,
});
</code></pre>

**Comment by techierishi** [comment]
<p>This solution worked. Thanks for you help.</p>

**Comment by mahesh-krishnakumar** [comment]
<p><a class="mention" href="/techierishi">@techierishi</a> Does this solution continue to work for you. I get a <code>500</code> response when I try to create the commit with <code>SHA1_EMPTY_TREE</code> as the tree.</p>

**Comment by mrchief** [comment]
<p>I’m facing the same. It seems to work for some repos but not all the time for every repo. I created a ticket for this <a href="https://github.community/t/500-error-while-creating-a-new-commit/203629" class="inline-onebox">500 error while creating a new commit</a></p>
